src/tyr/route_serializer_osrm.cc

```diff
@@ -1494,65 +1503,94 @@ json::ArrayPtr voice_instructions(const valhalla::DirectionsLeg::Maneuver* prev_
+    // the one hand distance_before_verbal_transition_alert_instruction has to be set, 
+    // so there is enough time to play the transition instruction afterwards. On the other hand
+    // there has to be enough time to play the upcoming verbal_pre_transition_instruction and 
+    // the verbal_post_transition_instruction itself. 
+    // The approximation here is that the verbal_post_transition_instruction takes 100
+    // meters to play + the 10 meters after the maneuver start which is added so that the
+    // instruction is not played directly on the intersection where the maneuver starts.
+    json::MapPtr voice_instruction_beginning = json::map({});
+    voice_instruction_beginning->emplace("distanceAlongGeometry", json::fixed_t{distance - 10, 1});
+    voice_instruction_beginning->emplace("announcement",
+                                          prev_maneuver->verbal_post_transition_instruction());
+    voice_instruction_beginning->emplace("ssmlAnnouncement",
+                                      "<speak>" + prev_maneuver->verbal_post_transition_instruction() + "</speak>"); 
+    voice_instructions_array->emplace_back(std::move(voice_instruction_beginning));
+  }
+
+  // If there is an alert instruction and we have enough time to play it, we will play it
+  // TODO: don't add alert instruction if there was already a pre/post transition aleart which is close by
+  if (!maneuver.verbal_transition_alert_instruction().empty() 
+    && distance_before_verbal_transition_alert_instruction > -1.0
+    && prev_maneuver->time() > SECONDS_BEFORE_VERBAL_TRANSITION_ALERT_INSTRUCTION + APPROXIMATE_VERBAL_PRERANSITION_SECONDS) {
     json::MapPtr voice_instruction_end = json::map({});
+    float distance_along_geometry = -1;
     if (maneuver_index == 1 && distance_before_verbal_transition_alert_instruction == distance) {
       // For the depart maneuver we want to play both the verbal_post_transition_instruction and
       // the verbal_transition_alert_instruction even if the maneuver is too short.
-      voice_instruction_end->emplace("distanceAlongGeometry", json::fixed_t{distance / 2, 1});
+      distance_along_geometry = distance / 2;
     } else {
       // In all other cases we use distance_before_verbal_transition_alert_instruction value
       // as it is capped to the maneuver length
-      voice_instruction_end
-          ->emplace("distanceAlongGeometry",
-                    json::fixed_t{distance_before_verbal_transition_alert_instruction, 1});
+      distance_along_geometry = distance_before_verbal_transition_alert_instruction;
     }
-    voice_instruction_end->emplace("announcement", maneuver.verbal_transition_alert_instruction());
+    //building voice instructions for the alert
+    float distance_km = (float) distance_along_geometry / 1000.0f; 
+    std::string instruction = narrative_builder->FormVerbalAlertApproachInstruction(distance_km, maneuver.verbal_transition_alert_instruction());
```

Comment:
Here is the part where I'm enhancing the alert instruction by the distance information. This needs the narrative_builder.